examples/notebooks/beam-ml/mltransform/compute_and_apply_vocab.ipynb

```diff
@@ -0,0 +1,430 @@
+      ],
+      "metadata": {
+        "id": "ZUSiAR62SgO8"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "[ComputeAndApplyVocabulary](https://beam.apache.org/releases/pydoc/current/apache_beam.ml.transforms.tft.html#apache_beam.ml.transforms.tft.ComputeAndApplyVocabulary) is a data processing transform that computes a unique vocabulary from a dataset and then maps each word or token to a distinct integer index. Use this transform to change textual data into numerical representations for machine learning tasks.\n",
+        "\n",
+        "Generating a vocabulary on the incoming dataset is a crucial preprocessing step while training machine learning modesl that deal with text data. By mapping words to numerical indices, the vocabulary reduces the complexity and dimensionality of dataset, allowing ML models to process the same words in a consistent way.\n",
+        "\n",
+        "This notebook shows how to use `MLTransform` to complete the following tasks:\n",
+        "* Use `write` mode in `MLTransform` to generate a vocabulary on the input text and assign an index value to each token.\n",
+        "* Use `read` mode to use the generated vocabulary and assign an index to a different dataset.\n",
+        "\n",
+        "`MLTransform` uses the `ComputeAndApplyVocabulary` transform, which is implemented by using `tensorflow_transform` to generate the vocabulary.\n",
+        "\n",
+        "For more information about using `MLTransform`, see [Preprocess data with MLTransform](https://beam.apache.org/documentation/ml/preprocess-data/) in the Apache Beam documentation"
+      ],
+      "metadata": {
+        "id": "Ebs9JtTrqQZH"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Install the required modules\n",
+        "\n",
+        "To use `ComputeAndVocabulary` with `MLTransfrom`, install `tensorflow_transform` and the Apache Beam SDK version 2.53.0 or later."
+      ],
+      "metadata": {
+        "id": "BnOMWq9C1Yo0"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "P3uUPotJo7Dc"
+      },
+      "outputs": [],
+      "source": [
+        "! git clone https://github.com/apache/beam.git\n",
+        "! cd beam/sdks/python\n",
+        "! pip install beam/sdks/python\n",
+        "! pip install tensorflow-transform --quiet"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import os\n",
+        "import tempfile\n",
+        "import apache_beam as beam\n",
+        "from apache_beam.ml.transforms.base import MLTransform\n",
+        "from apache_beam.ml.transforms.tft import ComputeAndApplyVocabulary"
+      ],
+      "metadata": {
+        "id": "e7-Vmawtq3xC"
+      },
+      "execution_count": 22,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "artifact_location = tempfile.mkdtemp(prefix='compute_and_apply_vocab_')\n",
```

Comment:
We should add a header above this section, something like: "Generate a vocabulary" and have a 1 sentence explanation of what we're going to do in this piece.